STM32U3: add DMA support

kbidani opened the pull request:
This PR adds DMA support and test for STM32U3.

kbidani committed:
tests: drivers: dma: loop_transfer: add stm32u3 boards overlays

Add DMA loop transfer test overlays for Nucleo U385RG-Q board.

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

kbidani committed:
boards: st: stm32u3: enable dma for stm32u3 boards

update documentation to indicate that DMA is
available on Nucleo U385RG-Q board.

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

kbidani committed:
dts: arm: st: u3: add gpdma

Add GPDMA support to STM32U3

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

etienne-lms reviewed (commented):
It would be better commit "tests: drivers: dma: loop_transfer: add stm32u3 boards overlays" comes after commit "dts: arm: st: u3: add gpdma" in your series.

mathieuchopstm commented on tests/drivers/dma/loop_transfer/boards/nucleo_u38rg_q.conf:
```diff
@@ -0,0 +1 @@
+CONFIG_DMA_LOOP_TRANSFER_CHANNEL_NR=8
```
Nit: `=0`/`=11` (first/last channel) would have a clearer meaning

kbidani committed:
tests: drivers: adc: adc_api: fix ADC DMA test for Nucleo U385RG-Q

Correct the ADC DMA test to ensure proper functionality on the
 Nucleo U385RG-Q board.

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

kbidani commented:
@teburd PTAL
